registry/client/repository.go

```diff
@@ -205,11 +205,30 @@ type tags struct {
 	name    reference.Named
 }
 
+type compatTagsURLKey struct{}
+
+func (compatTagsURLKey) String() string {
+	return "tags.url.compat"
+}
+
 // All returns all tags
 func (t *tags) All(ctx context.Context) ([]string, error) {
 	var tags []string
-
-	u, err := t.ub.BuildTagsURL(t.name)
+	var u string
+	var err error
+	var compatURL bool
+
+	// This is needed to be compatible with old registries that use "list" instead of "_list"
+	_, compatURL = ctx.Value(compatTagsURLKey{}).(bool)
```

Comment:
The way @dmcgowan suggested seems as less intrusive change to the original code so I changed my code that way. I am not sure how to go with "helper that takes url" idea for this case.